### PR TITLE
feat: defaultProvider

### PR DESCRIPTION
**Pull request by benjlevesque**

### CODE REVIEW

File: packages/payment-detection/src/provider.ts [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
 
-export const getDefaultProvider = (network?: string | ethers.providers.Network | undefined) => {
-  if (network === 'private') {
-    return new ethers.providers.JsonRpcProvider();
-  }
-  if (configuration.infura) {
-    return new ethers.providers.InfuraProvider(network, process.env.RN_INFURA_KEY);
+export const initPaymentDetectionProvider = (overrides: {
+  /** */
+  blockchainRpcs?: Record<string, string>;
```
Comment (vrolland) [inline packages/payment-detection/src/provider.ts:15]: `BlockchainRpcs` **might** be confusing as it's only for Ethereum forks.

File: packages/payment-detection/src/provider.ts [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
+): providers.Provider => {
+  if (typeof network === 'string') {
+    const rpc = blockchainRpcs[network];
+    if (rpc) {
+      return new providers.JsonRpcProvider(rpc);
+    }
   }
 
-  return ethers.getDefaultProvider(network);
+  return ethers.getDefaultProvider(network, { ...defaultProviderOptions, ...options });
```
Comment (vrolland) [inline packages/payment-detection/src/provider.ts:34]: Default of default can be hard to understand for new comers.
But I am personally comfortable with that.

File: packages/payment-detection/src/provider.ts [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
-let configuration = {
+let defaultProviderOptions: Record<string, string> = {
   // fallback to Ethers v4 default projectId
   infura: process.env.RN_INFURA_KEY || '7d0d81d0919f4f05b9ab6634be01ee73',
 };
 
-export const initPaymentDetectionProvider = (config: Partial<typeof configuration>) => {
-  configuration = { ...configuration, ...config };
+let blockchainRpcs: Record<string, string> = {
+  matic: 'https://rpc-mainnet.matic.network/',
```
Comment (vrolland) [inline packages/payment-detection/src/provider.ts:10]: missing other network like xDai & private ?

**Review by vrolland — APPROVED** [review]

File: packages/payment-detection/src/provider.ts (hunk 1 above)
Comment (benjlevesque) [reply packages/payment-detection/src/provider.ts:15]: forgot to comment, thx

File: packages/payment-detection/src/provider.ts (hunk 2 above)
Comment (benjlevesque) [reply packages/payment-detection/src/provider.ts:34]: you mean passing a param to `getDefaultProvider`? it's the way `ethers` does it. 

This wrapper enables NOT passing the param every time you call `getNetworkProvider`, but instead sharing the config

File: packages/payment-detection/src/provider.ts (hunk 3 above)
Comment (benjlevesque) [reply packages/payment-detection/src/provider.ts:10]: private, done. 
xdai & others will be added when relevant. This is only used for payment networks FYI

**Review by yomarion — APPROVED** [review]
Nice

**Review by vrolland — APPROVED** [review]
LGTM :)

**Pull request merged**